aten/src/ATen/core/dispatch/DispatchKeyExtractor.h

```diff
@@ -137,33 +137,58 @@ struct TORCH_API DispatchKeyExtractor final {
   void registerSchema(const FunctionSchema& schema) {
     TORCH_INTERNAL_ASSERT(dispatch_arg_indices_reverse_.is_entirely_unset());
     dispatch_arg_indices_reverse_ = makeBitsetForDispatchArgs(schema);
+    num_args_ = schema.arguments().size();
   }
   void deregisterSchema() {
     dispatch_arg_indices_reverse_ = c10::utils::bitset();
+    num_args_ = 0;
   }
 
   DispatchKeySet getDispatchKeySetBoxed(const torch::jit::Stack* stack) const {
     DispatchKeySet ks;
-    dispatch_arg_indices_reverse_.for_each_set_bit([&] (size_t reverse_arg_index) {
-      const auto& ivalue = torch::jit::peek(*stack, 0, reverse_arg_index + 1);
-      if (C10_LIKELY(ivalue.isTensor())) {
-        // NB: Take care not to introduce a refcount bump (there's
-        // no safe toTensorRef method, alas)
-        ks = ks | ivalue.unsafeToTensorImpl()->key_set();
-      } else if (C10_UNLIKELY(ivalue.isTensorList())) {
-        for (const at::Tensor& tensor : ivalue.toTensorList()) {
-          ks = ks | tensor.key_set();
+    if (C10_UNLIKELY(dispatch_arg_indices_reverse_.all())) {
```

Comment:
I'm not actually sure how expensive this new check is but I'm assuming pretty cheap?

Reply:
It's just a comparison on int64. The real potential problem is the branch.